lib/taurus/qt/qtgui/base/taurusbase.py

```diff
@@ -1243,6 +1243,48 @@ def __init__(self, name, parent=None, designMode=False):
         self.call__init__(TaurusBaseComponent, name,
                           parent=parent, designMode=designMode)
         self._setText = self._findSetTextMethod()
+        formatter = getattr(taurus.tauruscustomsettings,
```

Comment:
I had  the same dilemma when I implement it. I did it because `TaurusBaseComponent` has a programmatic API to set the "formater" (`setFormat()`) via python module or via python format (string). 
While the `formater`  only accept string (Something that only should come as a view input according my implementation purpose).

I did it to not mix both API after checking  that "all" (official) taurus widget inherit from `TaurusBaseWidget`. But there is not major reason to not move it

Reply:
The rule of thumb for me is: if an API makes sense when running with a QObject (e.g. by a QGraphicsObject) , then it should be in TaurusBaseComponent. If the API specifically uses QWidget calls, then it goes to TaurusBaseWidget.
I think that in your implementation, some of the methods belong to one case and some to the other